### PR TITLE
Implement len_trim and trim and remove old code

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/4593
Continuing: https://github.com/lfortran/lfortran/pull/3674

### CODE REVIEW

**Comment by certik** [comment]
Excellent, thanks for implementing this!

What are the issues that we need to fix to get this PR in?

**Review by Pranavchiku — COMMENTED** [review]
For Fortran backend failure you need to write correct function name instead of `StringTrim` in `asr_to_fortran.cpp` file and that should fix that failure.

**Comment by certik** [comment]
We either have to figure out a hack to get this working, which might be possible. If not, then we need to comment out the 3rd party code that uses it.

Either way is fine, as long as `len_trim/trim` is moved to the new intrinsics.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -902,6 +902,8 @@ RUN(NAME intrinsics_316 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) #
 RUN(NAME intrinsics_317 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # get_command_argument
 RUN(NAME intrinsics_318 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # cpu_time
 RUN(NAME intrinsics_319 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # system_clock
+RUN(NAME intrinsics_320 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # len_trim
```
Comment (gxyd) [inline integration_tests/CMakeLists.txt:905]: I checked it myself locally as well that this passes on LLVM >= 17 version as well, maybe add that label?

**Comment by Pranavchiku** [comment]
@gxyd check the following commit: https://github.com/HarshitaKalani/M_CLI2/commit/adaf33f4f82c08325ffa8d9563668c5d8880461f

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -902,6 +902,8 @@ RUN(NAME intrinsics_316 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) #
...
+RUN(NAME intrinsics_321 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # trim
```
Comment (gxyd) [inline integration_tests/CMakeLists.txt:906]: same label for this integration test as well

**Review by Pranavchiku — APPROVED** [review]

**Review by gxyd — COMMENTED** [review]
Otherwise, this looks good to me.

**Comment by HarshitaKalani** [comment]
Thanks for the review. Will do that with the next PR.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (certik) [reply integration_tests/CMakeLists.txt:905]: Which label? We don't have `llvm17` label anymore, just `llvm`.

Comment (certik) [reply integration_tests/CMakeLists.txt:905]: It's being tested with LLVM 17 and 18:
```
2024-10-04T09:05:24.6103078Z  596/1181 Test  #596: intrinsics_320 ...........................   Passed    0.01 sec
```

Comment (gxyd) [reply integration_tests/CMakeLists.txt:905]: Ah, I see. I didn't notice the commit that was merged yesterday, I guess it's all good then.

**Comment by gxyd** [comment]
> Thanks for the review. Will do that with the next PR.

There will no need for that I guess, consider Ondřej's comment here: https://github.com/lfortran/lfortran/pull/4938#discussion_r1787736005.

I've resolved those conversations as well.

**Comment by certik** [comment]
@HarshitaKalani great job! Thanks for implementing these.

**Pull request merged**